shared/chat/conversation/header-area/normal/container.tsx

```diff
@@ -59,26 +28,66 @@ const HeaderBranch = (props: Props) => {
 }
 
 export default Container.withSafeNavigation(
-  Container.connect(mapStateToProps, mapDispatchToProps, (stateProps, dispatchProps) => ({
-    badgeNumber: [...stateProps._badgeMap.entries()].reduce(
-      (res, [currentConvID, currentValue]) =>
-        // only show sum of badges that aren't for the current conversation
-        currentConvID !== stateProps._conversationIDKey ? res + currentValue : res,
-      0
-    ),
-    channelName: stateProps.channelName,
-    contactNames: stateProps._contactNames,
-    infoPanelOpen: stateProps.infoPanelOpen,
-    muted: stateProps.muted,
-    onBack: dispatchProps.onBack,
-    onOpenFolder: dispatchProps._onOpenFolder,
-    onShowProfile: dispatchProps.onShowProfile,
-    onToggleInfoPanel: dispatchProps.onToggleInfoPanel,
-    onToggleThreadSearch: dispatchProps.onToggleThreadSearch,
-    participants: stateProps._participants || [],
-    pendingWaiting: stateProps.pendingWaiting,
-    smallTeam: stateProps.smallTeam,
-    teamName: stateProps.teamName,
-    unMuteConversation: dispatchProps._onUnMuteConversation,
-  }))(HeaderBranch)
+  Container.connect(
+    (state, {infoPanelOpen, conversationIDKey}: OwnProps) => {
+      const meta = Constants.getMeta(state, conversationIDKey)
+      const _participants = meta.teamname ? null : meta.nameParticipants
+      const _contactNames = meta.participantToContactName
+
+      return {
+        _badgeMap: state.chat2.badgeMap,
+        _contactNames,
+        _conversationIDKey: conversationIDKey,
+        _participants,
+        channelName: meta.channelname,
+        infoPanelOpen,
+        muted: meta.isMuted,
+        pendingWaiting:
+          conversationIDKey === Constants.pendingWaitingConversationIDKey ||
+          conversationIDKey === Constants.pendingErrorConversationIDKey,
+        smallTeam: meta.teamType !== 'big',
+        teamName: meta.teamname,
+      }
+    },
+    (
+      dispatch: Container.TypedDispatch,
+      {safeNavigateUpPayload, onToggleInfoPanel, conversationIDKey}: OwnProps
+    ) => ({
+      _onOpenFolder: () => dispatch(Chat2Gen.createOpenFolder({conversationIDKey})),
+      _onUnMuteConversation: () =>
+        dispatch(Chat2Gen.createMuteConversation({conversationIDKey, muted: false})),
+      onBack: () => dispatch(safeNavigateUpPayload()),
+      onShowProfile: (username: string) => dispatch(createShowUserProfile({username})),
+      onToggleInfoPanel,
+      onToggleThreadSearch: () => dispatch(Chat2Gen.createToggleThreadSearch({conversationIDKey})),
+    }),
+    (stateProps, dispatchProps) => {
+      const visiblePath = getVisiblePath()
```

Comment:
Is this the best place to call it? It's not driven by `mapStateToProps`. I guess if user navigates away conversationIDKey would change, and we'd run `mergeProps` again and get the correct `visiblePath` here. Just thinking out loud here. If that sounds correct, maybe add a comment?

Reply:
It affects badge and badge is calculated here. It doesn't change given a mounted component. If it's elsewhere that place will have to have the badge and this Boolean